### PR TITLE
Exercise endpoints

### PR DESCRIPTION
**Pull request by Chulstro**
# SOM-Timer BE Pull Request Form

## Issues Addressed by this Pull Request
#27 

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
`New random exercise endpoint complete`

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Has This Been Tested?
- [ ] No
- [x] Yes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
`Checked status and filtering capability`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,23 @@
     "rest_interval": "7:00"
 }
 ```
+
+#### GET  '/api/rand_exercises'
+
+```
+#REQUEST BODY
```
Comment (sienna-kopf) [inline README.md:36]: YES documentation is the best!!

File: application/__init__.py [hunk 2]
```diff
@@ -31,9 +31,18 @@ def handle_error(e):
     api.prefix = '/api'
 
     from application.controllers.timers import TimersResource
-    from application.controllers.exercises import ExercisesResource
+    from application.controllers.exercises import ExercisesResource, RandExercises
```
Comment (sienna-kopf) [inline application/__init__.py:34]: So smart to make this RandExercises resource!

File: application/__init__.py [hunk 3]
```diff
@@ -31,9 +31,18 @@ def handle_error(e):
...
+
+    # @api.route("/rand_exercise")
+    # def rand_exercise():
+    #     category = request.args.get('category')
+    #     duration = request.args.get('duration')
+    #
+    #     exercises = Exercise.query.filter(Exercise.category==category, Exercise.duration==duration)
+    #     return random.choice(exercises)
```
Comment (sienna-kopf) [inline application/__init__.py]: @Chulstro remove this and push it up!

File: application/controllers/exercises.py [hunk 4]
```diff
@@ -38,6 +40,16 @@
     help='category parameter is required'
 )
 
+class RandExercises(Resource):
+    def get(self):
+        category = request.json['category']
```
Comment (sienna-kopf) [inline application/controllers/exercises.py:45]: smaahhhttt on the request.json! Duh that this is coming through in the request body!

File: application/controllers/exercises.py [hunk 5]
```diff
@@ -38,6 +40,16 @@
...
+        duration = request.json['duration']
+
+        exercise = Exercise.query.filter(Exercise.category == category, Exercise.duration == duration)\
+        .order_by(func.random()).first()
```
Comment (sienna-kopf) [inline application/controllers/exercises.py:49]: Now this looks like something I recognize!! Yay!

File: tests/test_exercises.py [hunk 6]
```diff
@@ -42,23 +42,28 @@ def test_get_all_exercises(self):
         self.assertEquals(payload['exercises'][1]['duration'], '15:00')
         self.assertEquals(payload['exercises'][1]['category'], 'SomaticCategory.MOVEMENT')
 
-    def test_create_exercise(self):
-        response = self.test_app.post(
-            '/api/exercises',
+    def test_find_random_exercise(self):
```
Comment (sienna-kopf) [inline tests/test_exercises.py:45]: AND a test! I love that once we get one thing up and running it is easy to transfer over!

**Review by sienna-kopf — APPROVED** [review]
TLDR @Chulstro is a genius

File: README.md (hunk 1 above)
Comment (sciencefixion) [reply README.md:36]: I'm never saving documentation till the end again!

File: application/controllers/exercises.py [hunk 7]
```diff
@@ -38,6 +40,16 @@
     help='category parameter is required'
 )
 
+class RandExercises(Resource):
```
Comment (sciencefixion) [inline application/controllers/exercises.py:43]: Love this use of a class. Perfect!

**Review by sciencefixion — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (rwilliams659) [reply README.md:36]: Yall rock for making this a priority - it's SO helpful for us!

File: application/controllers/exercises.py (hunk 5 above)
Comment (rwilliams659) [reply application/controllers/exercises.py:49]: Thanks for implementing the random functionality on the back-end! Makes it such a breeze for us

**Review by rwilliams659 — APPROVED** [review]
Thank you all for being so great! Can't wait to hook up the last endoint :)
